### DISCUSSION TITLE
Duplicate nodes returned in 2023-04 and unstable, fine in 2023-01

### DISCUSSION THREAD

**Discussion by duncan-fairley** [Shopify/storefront-api-feedback#182] [Help] [answered]
The following query, when called against `2023-04` and `unstable`:
```graphql
  query {
    metaobjects(first: 4, type: "shops") {
      nodes {
        id
      }
    }
  }
```
returns two of each metaobject (request ID: `14010230-6cba-4287-88d7-28d7d02587b1`):
```json
{
   "data": {
      "metaobjects": {
         "nodes": [
            {
               "id": "gid://shopify/Metaobject/24543329"
            },
            {
               "id": "gid://shopify/Metaobject/24543329"
            },
            {
               "id": "gid://shopify/Metaobject/24576097"
            },
            {
               "id": "gid://shopify/Metaobject/24576097"
            }
         ]
      }
   }
}
```
but returns correctly on `2023-01` (request ID: `d28799fb-e1e9-4e7a-a13a-de97b9f43391`)
```json
{
   "data": {
      "metaobjects": {
         "nodes": [
            {
               "id": "gid://shopify/Metaobject/24543329"
            },
            {
               "id": "gid://shopify/Metaobject/24576097"
            },
            {
               "id": "gid://shopify/Metaobject/24608865"
            },
            {
               "id": "gid://shopify/Metaobject/24641633"
            }
         ]
      }
   }
}
```

This is creating issues on a few live sites that are built to expect uniqueness

**Comment by frehner** [comment]
Hello @DuncanFa , I'm honestly not sure how frequently the Storefront API team checks this repo for bug reports. 

If you have what you believe to be a bug, would you mind reporting it in the Hydrogen repo instead? For bugs, we collaborate pretty closely with the Storefront API team to fix them. 

https://github.com/Shopify/hydrogen/issues

I only happened across your report here while I was searching for something else.

**Reply by duncan-fairley** [reply]
yes for sure! Thanks @frehner

**Reply by frehner** [reply, marked as answer]
Following the bug report here https://github.com/Shopify/hydrogen/issues/764
